background/usercss-helper.js

```diff
@@ -157,32 +138,20 @@
     });
   }
 
-  function install({url, direct, downloaded, tab}, sender = this.sender) {
-    tab = tab !== undefined ? tab : sender.tab;
-    url = url || tab.url;
-    if (direct && !downloaded) {
-      prefetchCodeForInstallation(tab.id, url);
-    }
-    return openURL({
-      url: '/install-usercss.html' +
-        '?updateUrl=' + encodeURIComponent(url) +
-        '&tabId=' + tab.id +
-        (direct ? '&direct=yes' : ''),
-      index: tab.index + 1,
-      openerTabId: tab.id,
-      currentWindow: null,
-    });
-  }
-
-  function prefetchCodeForInstallation(tabId, url) {
-    const key = TEMP_CODE_PREFIX + tabId;
-    tempCodeLastWriteDate = Date.now();
-    Promise.all([
-      download(url),
-      chromeLocal.setValue(key, {loading: true}),
-    ]).then(([code]) => {
-      chromeLocal.setValue(key, code);
-      setTimeout(() => chromeLocal.remove(key), TEMP_CODE_CLEANUP_DELAY);
+  window.openUsercssInstallPage = (tabId, url) =>
+    Promise.resolve(
+      url.startsWith('file:') ||
+      fetch(url, {method: 'HEAD'})
+        .then(r => /^text\/(css|plain)(;.*?)?$/i.test(r.headers.get('content-type'))))
+    .then(ok => ok && download(url))
```

Comment:
![image](https://user-images.githubusercontent.com/1324510/74689755-4eeb7280-5217-11ea-8c87-b8093741e20f.png)
`download` can't handle `file://` URLs in Firefox so we still need a content script to get the code from the tab.

Reply:
I wonder why does it work for me then? Gonna investigate.